### PR TITLE
Highlight field with error and fix anchors

### PR DESCRIPTION
**Pull request by slawosz**
https://regulatorydelivery.atlassian.net/browse/COSBETA-1209
https://regulatorydelivery.atlassian.net/browse/COSBETA-1210

The idea for anchors is following:
Instead of using different anchor and thus `id` for error messages, we could determine on which element in frontend (so probably only in `app/views/poison_centres/notifications/_date_filter_input.html.erb` we should place an `id`. This way it would be much easier to implement.

Review app:
https://cosmetics-pr-2211-search-web.london.cloudapps.digital/
https://cosmetics-pr-2211-submit-web.london.cloudapps.digital/

### CODE REVIEW

File: cosmetics-web/app/helpers/application_helper.rb [hunk 1]
```diff
@@ -43,4 +50,33 @@ def reference_number_for_display(notification)
   def wrap_summary_address(address_array)
     address_array.join("<span class=\"govuk-visually-hidden\">,</span><br />").html_safe
   end
+
+  # def error_id(search_form, attribute, part)
```
Comment (slawosz) [inline cosmetics-web/app/helpers/application_helper.rb:54]: I will get rid of this soon

File: cosmetics-web/app/models/gov_uk/date_from_form.rb [hunk 2]
```diff
@@ -0,0 +1,72 @@
+module GovUK
+  class DateFromForm
```
Comment (slawosz) [inline cosmetics-web/app/models/gov_uk/date_from_form.rb:2]: I spend some time trying to name it. The thing is, this should be located in `ActiveModel::Types::GovUKDate` but it does not work with Rails reloading, so its get called from there.

File: cosmetics-web/app/helpers/application_helper.rb [hunk 3]
```diff
@@ -43,4 +50,33 @@ def reference_number_for_display(notification)
...
+  #       attribute
+  #     end
+  #   elsif attr.is_a? Date
+  #     if search_form.errors[attribute].present? && part == :day
+  #       attribute
+  #     end
+  #   end
+  # end
+
+  def error_class(search_form, attribute, part)
```
Comment (scruti) [inline cosmetics-web/app/helpers/application_helper.rb:68]: does this helper have any specific dependency with search forms? `error_class` seems quite generic but `search_form` seems specific.

File: cosmetics-web/app/models/gov_uk/date_from_form.rb (hunk 2 above)
Comment (scruti) [reply cosmetics-web/app/models/gov_uk/date_from_form.rb:2]: Maybe `DateFrom` ? The `Form` suffix clashes with the form objects naming (and originally made me thought this was a form object :sweat_smile:

File: cosmetics-web/app/models/gov_uk/date_from_form.rb [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    IncompleteDate = Struct.new(:year, :month, :day) do
```
Comment (scruti) [inline cosmetics-web/app/models/gov_uk/date_from_form.rb:3]: Given that this got public methods and is not simply wrapping some attributes... maybe should be turned into a class?

**Review by scruti — APPROVED** [review]

**Pull request merged**